### ISSUE TITLE
Windows install/update does not always create/modify bin folder in appdata

### ISSUE THREAD

**Issue by damieng** [atom/atom#11450]
When the windows installer runs or the auto-updater executes the following files should be created/overwritten in c:\users<username>\appdata\local\atom
- atom
- atom.cmd
- apm
- apm.cmd

Each one acts as a simple shortcut to it's associated script in c:\users<username>\appdata\local\atom\app-1.x.y\resources\cli  (suffixless ones link to the .sh ones)

Various reports of this, using this bug to correlate them into a single issue. Doesn't always happen, need to find out why. Success/fail for these is not part of the squirrel log right now.

**Comment by TylerMoeller** [comment]
I just hit this issue today - all of my packages failed to update because apm could not be found. `apm.cmd` was still pointing at an old directory (app-1.5.1) that was no longer on my system. 

To fix the issue, I updated apm.cmd with a loop to find the latest version:

```batch
@echo off
for /f "tokens=*" %%I in ('dir /b /ad %LOCALAPPDATA%\atom\app*') do set appdir=%%I
"%~dp0..\%appdir%\resources\app\apm\bin\apm.cmd" %*
```

Also did the same with atom.cmd to keep my shortcuts working across client updates:

```batch
@echo off
for /f "tokens=*" %%I in ('dir /b /ad %LOCALAPPDATA%\atom\app*') do set appdir=%%I
"%~dp0..\%appdir%\resources\cli\atom.cmd" %*
```

**Comment by damieng** [comment]
Yeah those are nice scripts. If I can't get definitively to the bottom of why 'bin' isn't getting updated reasonably quick I might do something similar to mitigate.

**Comment by hobbsd** [comment]
My Windows 7 install of Atom 1.8.0 did create the `bin` scripts, but my Windows 10 install didn't, so I had to add `%LOCALAPPDATA%\atom\app-1.8.0\` to my `PATH` and use `atom.exe` from the commandline.

**Comment by damieng** [comment]
@hobbsd can you type in the command line;

```atom --squirrel-updated```

And see if that creates the shell/batch files in `bin` ?

**Comment by wasifhossain** [comment]
@damieng indeed running the command creates those shell/batch files in `bin`. Tested on Windows 10.

**Comment by damieng** [comment]
Definitely need to get some logging into that so we can troubleshoot when it fails.

**Comment by hobbsd** [comment]
Yes, that command works for me as well on Windows 10.

**Comment by wasifhossain** [comment]
Off topic... I used to be a Microsoft stack developer using tools like Visual Studio for 4/5 years which eventually led me to start using VS Code editor last April. But as I tried to enter the open source horizon very recently and get rid of the MS stack, I have eventually found Atom as the best open source development editor (feeling at home after migrating from Visual Studio). Great work :+1:  Atom team

Note: Sublime Text is also very good. But I fell in affection with Atom ;)

**Comment by aisa0906** [comment]
My Windows 7 install of Atom 1.12.7 did not create the bin files. What do you recommend?

**Comment by rsese** [comment]
From https://github.com/atom/atom/issues/16831:

> Verify that you can still reproduce the issue in the latest version of Atom
> Comment that the issue is still reproducible and include:
> What version of Atom you reproduced the issue on

1.24

> What OS and version you reproduced the issue on

Windows 8.1

> What steps you followed to reproduce the issue

This was on an install @MandarSant or an update?

**Comment by MandarSant** [comment]
I have this issue as well as following issues
[#973](https://github.com/atom/github/issues/973)
[#14446](https://github.com/atom/atom/issues/14446)

**Comment by MandarSant** [comment]
@rsese it is on both install and update

**Comment by alex-robbins** [comment]
A coworker of mine just saw this when installing Atom 1.30 on a Windows 10 machine for the first time.

**Comment by alex-robbins** [comment]
Just noticed that I had an old `bin` folder at `%ProgramData%/%USERNAME%/atom/bin`. The files in there referenced an old, no longer installed version of atom (1.29.0). The directory was still on my PATH.

Installing a new beta version (1.32.0-beta2) updated (or maybe created) files in `%LOCALAPPDATA%/atom/bin` so they referenced 1.32.0-beta2.

So in addition to the directory sometimes not being updated, there's an old one that's not always cleaned up. Not sure if that's a separate issue or not.

**Comment by david-mears** [comment]
"atom" is not a command recognised from command line in Windows 10.0.17134. There is no atom/bin file to be seen. Using latest Atom version: Atom 1.32.2

**Comment by Neustradamus** [comment]
I would like to install in C:\Program Files\ for the current user or for all users.
How it is possible?

**Comment by Neustradamus** [comment]
@damieng: It is solved?

**Comment by Neustradamus** [comment]
All good softwares when we install use by default "Program Files" (or recently "Program Files (x86)" for 32 bit apps in 64 bit system).
Bad softwares do not use it.

It is since Windows 95 and maybe before, no?

**Comment by pt-dev** [comment]
> Verify that you can still reproduce the issue in the latest version of Atom
> Comment that the issue is still reproducible and include:
> What version of Atom you reproduced the issue on
1.44.0
> What OS and version you reproduced the issue on
Windows 10 
![image](https://user-images.githubusercontent.com/3056147/75696973-69433880-5c7a-11ea-9151-9b3506f9dce5.png)

> What steps you followed to reproduce the issue
Installing Atom with the official installer.

**Comment by mickey203** [comment]
Atom is a good editor. Why make it so complicated to install. Why would anyone install the same exe file  n number of times
It is invisible once you install it. If at all it is installable for Windows then it should follow exe norms

**Comment by manngo** [comment]
> When the windows installer runs or the auto-updater executes the following files should be created/overwritten in c:\users\appdata\local\atom

I have the same problem and what looks like a working solution here:

https://discuss.atom.io/t/atom-bin-folder-is-not-always-installed/81279

TLDR:

	%userprofile%\AppData\Local\atom\atom.exe --squirrel-install

will seems to install the folder.
